### PR TITLE
Chat and menu fixes

### PR DESCRIPTION
**Pull request by aduermael**
- Fix text input focus/unfocus conditions
- Hide message dates within non-fullscreen chat console
- Better handle ESC key (use back button in active modal or close, unfocus chat input)

### CODE REVIEW

File: lua/modules/menu.lua [hunk 1]
```diff
@@ -100,6 +101,19 @@ function closeModal()
 	end
 end
 
+-- pops active modal content
+-- closing modal when reaching root content
+function popModal()
+	if activeModal == nil then
+		return
+	end
```
Comment (XLegland) [inline lua/modules/menu.lua:109]: Wouldn't it be better to call `closeModal` here? `if activeModal ~= nil and #activeModal.contentStack > 1 then ...`

**Review by XLegland — COMMENTED** [review]
If I press `/` on the hub it opens the chat but does not input `/`

File: lua/modules/menu.lua (hunk 1 above)
Comment (aduermael) [reply lua/modules/menu.lua:109]: If activeModal is nil, there's nothing to close or pop.
I like the if + early return design for clarity.

**Comment by aduermael** [comment]
> If I press `/` on the hub it opens the chat but does not input `/`

@XLegland Thanks, good catch, I fixed it!

**Review by XLegland — APPROVED** [review]
That's good to me 👍

**Pull request merged**